### PR TITLE
Devkit Migration Tool

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: mule-sdk/v/1.1/dmt.adoc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+NOTE: Take in mind that test classes are not going to be ran, migrated nor copied (because it won't compile using the SDK).
+
+== The Generated Extension
+
+It's *very* important to understand that the generated extension may not compile nor work as expected
+right away after being generated, *manual work may be required to make the extension work properly*.
+
+The "spirit" of mule 4 is different than the mule 3 one and some things that were supported or allowed before
+aren't anymore - e.g. accessing the mule event or mule message -. This make the migration in some case impossible and that
```
Comment (ndinu) [inline mule-sdk/v/1.1/dmt.adoc]: This makes the migration impossible in some cases or the migrated components may not follow

File: mule-sdk/v/1.1/dmt.adoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+the migrated components doesn't follow best practices that MuleSoft tries to enforce for SDK users.
+
+For cases like this the DMT is smart enough to mark the code where a change is required with links to the documentation
+where you should find the way to make it work, but you should keep in mind that there are things (e.g changing, creating
+or getting variables) that you simply *won't* be able to do it anymore, and you should change that to
```
Comment (ndinu) [inline mule-sdk/v/1.1/dmt.adoc]: be able to do it anymore -> be able to do anymore
to -> too

File: mule-sdk/v/1.1/dmt.adoc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+All the processors will be migrated to Operations, all within the same class.
+Parameters of the Processor should be reflected in the Operation parameters and
+also elements as Config or injected fields present in the old Connector that should
+be setup will be passed in as parameters.
+
+see: link:operations[SDK Operations Documentation Reference] to learn more about Extension Operations.
+
+=== Sources
+
+Both polling and triggered sources will be migrated using the Polling Sources SDK feature
```
Comment (ndinu) [inline mule-sdk/v/1.1/dmt.adoc]: Triggered sources are also migrated using the Polling Sources SDK feature?

File: mule-sdk/v/1.1/dmt.adoc [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+and the Mule 4 Scheduler Service respectively.
+
+As default Sources code will compile and run, but comments will be added to the generated sources classes
```
Comment (ndinu) [inline mule-sdk/v/1.1/dmt.adoc]: As default Sources -> As default, Sources

File: mule-sdk/v/1.1/dmt.adoc [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+@ConnectionManagement strategies will be migrated to a PoolingConnectionProvider, so the generated
+connections are pooled. A cache of connections is maintained so that they can be reused when future requests require one.
+
+If the @ConnectionManagement connect method is marked as *Single Instance* are going to be migrated to a CachedConnectionProvider, like a Configuration.
+
+=== Also Supported
+
+* Paged Operations
+* Metadata Classes (MetaData Objects are not migrated, manual migration to the MetadataTypes is required)
```
Comment (ndinu) [inline mule-sdk/v/1.1/dmt.adoc]: Should we add a link to SDK Metadata Docs?

File: mule-sdk/v/1.1/dmt.adoc [hunk 6]
```diff
@@ -0,0 +1,116 @@
+= DevKit Migration Tool (Beta Release)
+:keywords: mule, sdk, devkit, migration, migrate, connector
+
+The DevKit Migration Tool (DMT) serves as a starting point for migrating 
+from a Connector built with Devkit that runs on Mule 3.x to the SDK
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: ~from~ `a Mule 3 Connector built with Devkit to a new Mule 4 Connector built using Mule 4's SDK` (link "SDK" to SDK docs)

File: mule-sdk/v/1.1/dmt.adoc [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+model of Mule 4.
+
+The DMT introspects the DevKit connector code and generates the source Java code
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: introspects your Connector based on Devkit, generating new source code that renders it compatible with the new SDK model.

File: mule-sdk/v/1.1/dmt.adoc [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+of a new SDK extension. The new extension code will wrap the current connector code
+(as it would any other Java library) and delegate responsibility for the code 
+to the already-tested connector code.
+
+== Executing the DMT
+
+The DMT must be executed from the command line. Before executing it, you need
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: In order to execute the DMT, all you need to do is go to the connector's `pom.xml` file and change the `parent` artifact from this: asdasd to this: dml parent.

Now that the `parent` has been changed, just build your connector with a `mvn clean package`

File: mule-sdk/v/1.1/dmt.adoc [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+[source, xml, linenums]
+----
+<parent>
+  <groupId>org.mule.tools.devkit</groupId>
+  <artifactId>mule-devkit-parent</artifactId>
+  <version>3.9.4</version>
+</parent>
+----
+
+To execute the DMT, you must move your current version to `4.0.0`.
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: not

File: mule-sdk/v/1.1/dmt.adoc [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+Now, from the your connector's root directory, you can execute the following line
+to start the migration:
+
+----
+mvn clean package
+----
+
+Once the process finishes, you should see a `Build Success` message, and you will find the generated
+extension project under the `_{rootdir}/target/generated-sources/extension_` folder.
+
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: maybe screenshot of the resulting project tree?

File: mule-sdk/v/1.1/dmt.adoc [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+NOTE: Test classes are not going to be run, migrated, or copied (because it won't compile using the SDK).
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: was there a flag for this? Don't mention the compilation thing.

File: mule-sdk/v/1.1/dmt.adoc [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+== The Generated Extension
+
+It's *very* important to understand that the generated extension might not compile or work as expected
+right after it is generated. *Manual work might be required to make the extension work properly*.
+
+Mule 4 is different from Mule 3. Some things that were supported or allowed in Mule 3 (such as accessing the mule event or mule message) are not in Mule 4. These changes can make it impossible to migrate some components, and some components that do migrate might fail to follow the best practices that MuleSoft tries to enforce for SDK users.
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: impossible to automatically migrate

File: mule-sdk/v/1.1/dmt.adoc [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+
+Both @Configuration and @ConnectionManagement Strategies are migrated.
+
+See: link:connections[Connections in the SDK] to learn more about connections.
+
+==== Configuration
+
+@Configuration strategies are going to be migrated to a CachedConnectionProvider, which
+will provide a single connection instance for all operations until that connection is stopped.
+
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: link cached cp

File: mule-sdk/v/1.1/dmt.adoc [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+==== Connection Management
+
+@ConnectionManagement strategies will be migrated to a PoolingConnectionProvider, so the generated
+connections are pooled. A cache of connections is maintained so that they can be reused when future requests require one.
+
+If the @ConnectionManagement connect method is marked as *Single Instance* are going to be migrated to a CachedConnectionProvider, like a Configuration.
+
```
Comment (alegmarra) [inline mule-sdk/v/1.1/dmt.adoc]: link to pooling and cached cp

**Pull request merged**